src/unix/mod.rs

```diff
@@ -16,14 +16,17 @@ pub type cc_t = ::c_uchar;
 pub enum DIR {}
 pub enum locale_t {}
 
+#[cfg(not(target_os = "l4re"))]
```

Comment:
We typically prefer to avoid these sorts of `#[cfg]` annotations in libc wherever possible, mind pushing these structs further down the hierarchy to skip them on l4re?